File: gm-utils/js/src/commom/utils/string/StringLength.js
import IntegerCompare from '../integer/IntegerCompare';
import Null from '../object/Null';

export default class StringLength {

	static get(s) {
		if (Null.is(s)) {
			return 0;
		}
		return s.length;
	}

	static is(s, size) {
		return IntegerCompare.eq(StringLength.get(s), size);
	}

	static max(s, max) {
		if (StringLength.get(s) > max) {
			return s.substring(0, max);
		}
		return s;
	}

}

File: gm-utils/src/main/java/src/commom/utils/string/StringLength.java
package src.commom.utils.string;

import src.commom.utils.integer.IntegerCompare;
import src.commom.utils.object.Null;

public class StringLength {

	private StringLength() {}

	public static int get(String s) {
		if (Null.is(s)) {
			return 0;
		}
		return s.length();
	}

	public static boolean is(String s, int size) {
		return IntegerCompare.eq(get(s), size);
	}

	public static String max(String s, int max) {
		
		if (StringEmpty.is(s)) {
			return "";
		}
		
		if (StringLength.get(s) > max) {
			return s.substring(0, max);
		}
		
		return s;
		
	}

}
